src/coprocessor/dag/expr/builtin_op.rs

```diff
@@ -70,19 +73,32 @@ impl FnCall {
     }
 
     pub fn unary_minus_int(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
-        unimplemented!()
+        let val = try_opt!(self.children[0].eval_int(ctx, row));
+        if mysql::has_unsigned_flag(self.tp.get_flag() as u64) {
```

Comment:
The function you reference is unrelated to the check here. `self.tp` only provides the field type information about the result returned by this function, not the values of children.